propagation_test.go

```diff
@@ -148,6 +148,41 @@ func TestBaggagePropagationHTTP(t *testing.T) {
 	assert.Equal(t, map[string]string{"some-key": "98:765"}, sp2.(SpanContext).baggage)
 }
 
+func TestJaegerBaggageHeader(t *testing.T) {
+	tracer, closer := NewTracer("hobbits", NewConstSampler(true), NewNullReporter())
+	defer closer.Close()
+
+	h := http.Header{}
+	h.Add(JaegerBaggageHeader, "key1=value1, key 2=value two")
+
+	sp2, err := tracer.Extract(opentracing.HTTPHeaders, opentracing.HTTPHeadersCarrier(h))
+	require.NoError(t, err)
+
+	assert.Equal(t, map[string]string{"key1": "value1", "key 2": "value two"}, sp2.(SpanContext).baggage)
```

Comment:
this is not the complete test of the new logic. You need to `s/sp2/ctx` and pass it to `ChildOf(ctx)` when creating a new span. Then the assertion should be of the form
```
assert.Equal(t, "value1", span.BaggageItem("key1"))
```
This ensures that the span is started with the right baggage. You can also validate `ctx.IsContainerOnly`